actors/init/src/main/java/org/sunbird/init/actors/SystemInitActor.java

```diff
@@ -0,0 +1,226 @@
+    Util.DbInfo orgDbInfo = Util.dbInfoMap.get(JsonKey.ORG_DB);
+    validateChannel(req);
+    req.put(JsonKey.CREATED_BY, JsonKey.INITIALISER);
+    String uniqueId = ProjectUtil.getUniqueIdFromTimestamp(actorMessage.getEnv());
+    req.put(JsonKey.ID, uniqueId);
+    req.put(JsonKey.ROOT_ORG_ID, uniqueId);
+    req.put(JsonKey.HASHTAGID, uniqueId);
+    req.put(JsonKey.CREATED_DATE, ProjectUtil.getFormattedDate());
+    req.put(JsonKey.STATUS, ProjectUtil.OrgStatus.ACTIVE.getValue());
+    req.put(JsonKey.IS_ROOT_ORG, true);
+    req.put(JsonKey.IS_DEFAULT, true);
+
+    String slug = Slug.makeSlug((String) req.get(JsonKey.CHANNEL), true);
+    boolean bool = isSlugUnique(slug);
+    if (bool) {
+      req.put(JsonKey.SLUG, slug);
+    } else {
+      ProjectCommonException.throwClientErrorException(
+          ResponseCode.slugIsNotUnique, ResponseCode.slugIsNotUnique.getErrorMessage());
+    }
+
+    boolean isChannelRegistered = Util.registerChannel(req);
+    if (!isChannelRegistered) {
+      ProjectCommonException.throwServerErrorException(
+          ResponseCode.channelRegFailed, ResponseCode.channelRegFailed.getErrorMessage());
+    }
+
+    Organization org = mapper.convertValue(req, Organization.class);
+    req = mapper.convertValue(org, Map.class);
+    Response result =
+        cassandraOperation.insertRecord(orgDbInfo.getKeySpace(), orgDbInfo.getTableName(), req);
+    addInitialisationFlagToSystemSettings();
+    ProjectLogger.log("Org data saved into cassandra.");
+    ProjectLogger.log("Created org id is ----." + uniqueId);
+    result.getResult().put(JsonKey.ORGANISATION_ID, uniqueId);
+    sender().tell(result, self());
+    Request orgReq = new Request();
+    orgReq.getRequest().put(JsonKey.ORGANISATION, req);
+    orgReq.setOperation(ActorOperations.INSERT_ORG_INFO_ELASTIC.getValue());
+    ProjectLogger.log("Calling background job to save org data into ES" + uniqueId);
+    tellToAnother(orgReq);
+  }
+
+  /**
+   * This method checks if the given channel slug value is unique
+   *
+   * @param slug input slug value need to checked.
+   * @return returns true if slug unique or return false
+   */
+  private boolean isSlugUnique(String slug) {
+    if (StringUtils.isNotBlank(slug)) {
+      Map<String, Object> filters = new HashMap<>();
+      filters.put(JsonKey.SLUG, slug);
+      filters.put(JsonKey.IS_ROOT_ORG, true);
+      Map<String, Object> esResult =
+          elasticSearchComplexSearch(
+              filters, EsIndex.sunbird.getIndexName(), EsType.organisation.getTypeName());
+      if (isNotNull(esResult)
+          && esResult.containsKey(JsonKey.CONTENT)
+          && isNotNull(esResult.get(JsonKey.CONTENT))) {
+        return (((List) esResult.get(JsonKey.CONTENT)).isEmpty());
+      }
+    }
+    return false;
+  }
+
+  /** This method writes the rootOrgInitialised status(true) to system settings */
+  private void addInitialisationFlagToSystemSettings() {
```

Comment:
This method swallows exceptions in writing the initialised flag. The caller will then assume that the initialised flag has been written and will return a successful result. However, the flag will not have been written. Need to revisit this behaviour.